Sources/NIO/EventLoop.swift

```diff
@@ -229,6 +229,9 @@ public protocol EventLoop: EventLoopGroup {
     /// Submit a given task to be executed by the `EventLoop`
     func execute(_ task: @escaping () -> Void)
 
+    /// Execute the given function and synchronously complete the given `EventLoopPromise` (if not `nil`).
+    func executeAndComplete<T>(_ promise: EventLoopPromise<T>?, _ body: () throws -> T)
```

Comment:
I don't think this realistically belongs on the `EventLoop`, this method probably belongs to `EventLoopPromise`.

Reply:
@Lukasa @weissi I'm personally feel this method belongs to EventLoop other than EventLoopPromise, that's the reason I tried to create this PR#1747 to replace PR#1746. Maybe I can dig into the Netty code to get more ideas to help you guys to make a decision.

Reply:
The reason I don't think it does is that the implementation you provide does not use `self` anywhere within it. It does not rely on the event loop in any way.

Reply:
@Lukasa I agree with you, it doesn't rely on event loop. I'm going ahead to close this PR.

I had also checked out the Netty's code, found its `bind` method and how it updates the promise. So it looks like there is no counterpart for `func executeAndComplete()` in Netty, it is doing to same thing inline. I'm not an expert of Netty, I hope I found the right place.

Anyway, I think `func executeAndComplete()` is only about executing the wrapped code, catching exception and then updating the promise correspondingly, there is nothing about EventLoop. Sorry for the confusion, and I'm going to close this PR.

![image](https://user-images.githubusercontent.com/354776/111176080-65c9d900-857f-11eb-8640-d24d478c599a.png)

cc @weissi